### PR TITLE
Solving no var violation by changing 'var' to 'let' in javascripts/main.js 

### LINKED ISSUE #1513: resolve no-var violations in javascripts/main.js
We have some very old ESLint rules that I'd like to clean up, and our use of `var` is one of them. ES6 introduces `let` and `const` as a way to make clearer when a variable can be used, and I'd like to reduce our use of `var` where we don't need it:

If you know a bit about JS you can help out by following these steps:

 - create a new branch from `gh-pages`
 - in `javascripts/main.js` add this comment at the top of the file, before any source code:

```js
/* eslint no-var: [ "error" ] */
```

 - in a terminal, run `npm run lint`, which should emit a bunch of violations:

```
$ npm run lint

> up-for-grabs.net@1.0.0 lint /Users/shiftkey/src/up-for-grabs.net
> eslint javascripts/*.js tests/**/*.js


/Users/shiftkey/src/up-for-grabs.net/javascripts/main.js
   15:3   error  Unexpected var, use let or const instead  no-var
   18:3   error  Unexpected var, use let or const instead  no-var
   ...
```

 - run `npm run lint-fix` to fix whatever can be automatically tidied up
 - review and address the remaining violations manually
 - run `npm run prettier-fix` to ensure the formatting of the changes is fine
 - [test the site locally](https://github.com/up-for-grabs/up-for-grabs.net#testing-the-site-locally)
 - commit the changes and open a PR
 - test out the Netlify deploy site (click the `Details` link) and confirm the client-side functionality is still working

### PR DESCRIPTION
**Pull request by XxZhang2017**
Resolves #1513

### CODE REVIEW

File: javascripts/main.js [hunk 1]
```diff
@@ -210,29 +211,29 @@ define([
...
    */
-  var prepareForHTML = function(text) {
+  const prepareForHTML = function(text) {
     return text ? text.toLowerCase().split(',') : text;
   };
 
-  var issueCount = function(project) {
-    var a = $(project).find('.label a'),
+  const issueCount = function(project) {
+    const a = $(project).find('.label a'),
```
Comment (shiftkey) [inline javascripts/main.js]:
```suggestion
    const a = $(project).find('.label a');
```

File: javascripts/main.js [hunk 2]
```diff
@@ -210,29 +211,29 @@ define([
...
       gh = a
```
Comment (shiftkey) [inline javascripts/main.js]:
```suggestion
      const gh = a
```

File: javascripts/main.js [hunk 3]
```diff
@@ -210,29 +211,29 @@ define([
...
         .attr('href')
         .match(/github.com(\/[^\/]+\/[^\/]+\/)(?:issues\/)?labels\/([^\/]+)$/),
```
Comment (shiftkey) [inline javascripts/main.js]:
```suggestion
        .match(/github.com(\/[^\/]+\/[^\/]+\/)(?:issues\/)?labels\/([^\/]+)$/);
```

File: javascripts/main.js [hunk 4]
```diff
@@ -210,29 +211,29 @@ define([
...
-      count = a.find('.count');
+      counthref = a.find('.count');
```
Comment (shiftkey) [inline javascripts/main.js]:
```suggestion
      let count = a.find('.count');
```

File: javascripts/main.js [hunk 5]
```diff
@@ -210,29 +211,29 @@ define([
...
 
-    if (count.length) {
+    if (counthref.length) {
```
Comment (shiftkey) [inline javascripts/main.js]:
```suggestion
    if (count.length) {
```

File: javascripts/main.js [hunk 6]
```diff
@@ -210,29 +211,29 @@ define([
...
       return;
     }
 
     if (!gh) {
-      count = $(
+      countspantitle = $(
```
Comment (shiftkey) [inline javascripts/main.js]:
```suggestion
      count = $(
```

File: javascripts/main.js [hunk 7]
```diff
@@ -210,29 +211,29 @@ define([
...
         '<span class="count" title="Issue count is only available for projects on GitHub.">?</span>'
       ).appendTo(a);
       return;
     }
 
-    count = $(
+    countspanimg = $(
```
Comment (shiftkey) [inline javascripts/main.js]:
```suggestion
    count = $(
```

File: javascripts/main.js [hunk 8]
```diff
@@ -241,20 +242,20 @@ define([
 
     fetchIssueCount(ownerAndName, labelEncoded).then(
       resultCount => {
-        count.html(resultCount);
+        countspanimg.html(resultCount);
```
Comment (shiftkey) [inline javascripts/main.js]:
```suggestion
        count.html(resultCount);
```

File: javascripts/main.js [hunk 9]
```diff
@@ -241,20 +242,20 @@ define([
...
       },
       error => {
         const message = error.message ? error.message : error;
-        count.html('?!');
-        count.attr('title', message);
+        countspanimg.html('?!');
```
Comment (shiftkey) [inline javascripts/main.js]:
```suggestion
        count.html('?!');
```

File: javascripts/main.js [hunk 10]
```diff
@@ -241,20 +242,20 @@ define([
...
+        countspanimg.attr('title', message);
```
Comment (shiftkey) [inline javascripts/main.js]:
```suggestion
        count.attr('title', message);
```

**Review by shiftkey — CHANGES_REQUESTED** [review]
This all looks good aside from the middle, because we'd put `count` (which may update based on what it finds) alongside two other variables that don't need to change.

I think the suggested changes should do it, but we might need another round of review after applying those.

**Comment by XxZhang2017** [comment]
@shiftkey The reason that I change it is as follow. The pre-const rules failed it if issueCount, a, gh are let. It suggests changing to const. I changed them. Then ESlint show the error message that is about reassigning value to the const. I have to modify again. Any other suggestions to do it I am happy to think about it. 
` count = a.find('.count');

    if (count.length) {
      return;
    }

    if (!gh) {
      count = $(
        '<span class="count" title="Issue count is only available for projects on GitHub.">?</span>'
      ).appendTo(a);
      return;
    }

    count = $(
      '<span class="count"></span>'
    ).appendTo(a);

    const ownerAndName = gh[1];
    const labelEncoded = gh[2];

    fetchIssueCount(ownerAndName, labelEncoded).then(
      resultCount => {
        count.html(resultCount);
      },
      error => {
        const message = error.message ? error.message : error;
        count.html('?!');
        count.attr('title', message);
      }`

**Comment by shiftkey** [comment]
> Any other suggestions to do it I am happy to think about it.

We need to refactor this single `var` into three `var` statements, and restore the names to what they were before: 

```js
const issueCount = function(project) {
    var a = $(project).find('.label a');
    var gh = a
      .attr('href')
      .match(/github.com(\/[^\/]+\/[^\/]+\/)(?:issues\/)?labels\/([^\/]+)$/);
    var count = a.find('.count');

    if (count.length) {
      return;
    }

    if (!gh) {
      count = $(
        '<span class="count" title="Issue count is only available for projects on GitHub.">?</span>'
      ).appendTo(a);
      return;
    }

    count = $(
      '<span class="count"><img src="images/octocat-spinner-32.gif" /></span>'
    ).appendTo(a);

    const ownerAndName = gh[1];
    const labelEncoded = gh[2];

    fetchIssueCount(ownerAndName, labelEncoded).then(
      resultCount => {
        count.html(resultCount);
      },
      error => {
        const message = error.message ? error.message : error;
        count.html('?!');
        count.attr('title', message);
      }
    );
  };
``` 

Now, when you run `npm run lint-fix` it'll detect that the first two are unused (changing them to `const`) and the last one is updated (changing it to `let`):

```js
const issueCount = function(project) {
    const a = $(project).find('.label a');
    const gh = a
      .attr('href')
      .match(/github.com(\/[^\/]+\/[^\/]+\/)(?:issues\/)?labels\/([^\/]+)$/);
    let count = a.find('.count');

    if (count.length) {
      return;
    }

    if (!gh) {
      count = $(
        '<span class="count" title="Issue count is only available for projects on GitHub.">?</span>'
      ).appendTo(a);
      return;
    }

    count = $(
      '<span class="count"><img src="images/octocat-spinner-32.gif" /></span>'
    ).appendTo(a);

    const ownerAndName = gh[1];
    const labelEncoded = gh[2];

    fetchIssueCount(ownerAndName, labelEncoded).then(
      resultCount => {
        count.html(resultCount);
      },
      error => {
        const message = error.message ? error.message : error;
        count.html('?!');
        count.attr('title', message);
      }
    );
  };
```

**Comment by XxZhang2017** [comment]
@shiftkey Thank you for the explaining. I committed  the changes. Is everything ok now ?

File: javascripts/main.js [hunk 11]
```diff
@@ -210,16 +211,16 @@ define([
...
-    var a = $(project).find('.label a'),
-      gh = a
+  const issueCount = function(project) {
+    const a = $(project).find('.label a');
+      const gh = a
         .attr('href')
-        .match(/github.com(\/[^\/]+\/[^\/]+\/)(?:issues\/)?labels\/([^\/]+)$/),
-      count = a.find('.count');
+        .match(/github.com(\/[^\/]+\/[^\/]+\/)(?:issues\/)?labels\/([^\/]+)$/);
+      counthref = a.find('.count');
```
Comment (shiftkey) [inline javascripts/main.js]:
```suggestion
      let count = a.find('.count');
```

**Review by shiftkey — CHANGES_REQUESTED** [review]
Almost there - just one more change and then run `npm run prettier-fix` and commit the cleanup.

**Comment by XxZhang2017** [comment]
@shiftkey I think there should not be any error this time.

**Comment by shiftkey** [comment]
@XxZhang2017 you're close - just need to do this bit:

> then run `npm run prettier-fix` and commit the cleanup.

**Comment by XxZhang2017** [comment]
@shiftkey Hi, may I ask a question ?  I have trouble synchronize the fork project with the original one.

**Comment by XxZhang2017** [comment]
@ shiftkey I am sorry that there is issue. The picture shows what I did. I think it should be fine.
Screen Shot 2019-10-14 at 11 26 30 PM

**Comment by XxZhang2017** [comment]
@ shiftkey I am sorry that there is still issue. But I have run it. I have no idea what is wrong with it.

Screen Shot 2019-10-14 at 11 26 30 PM
It passed the check here. 
Screen Shot 2019-10-15 at 12 28 59 AM
Anything I can do at this point ?

**Comment by shiftkey** [comment]
@XxZhang2017 ensure you are running the command in your `edit-ESLint` branch (screenshot seems to suggest you're on the `gh-pages` branch)

**Review by shiftkey — APPROVED** [review]

**Comment by shiftkey** [comment]
@XxZhang2017 thanks!

**Comment by shiftkey** [comment]
@all-contributors please add @XxZhang2017 for code

**Pull request merged**